cmake/modules/FindXercesC.cmake

```diff
@@ -54,8 +54,10 @@ find_path(
 
 # Find the xerces libraries
 if (NOT XercesC_LIBRARIES)
-    find_library(XercesC_LIBRARY_RELEASE NAMES xerces-c xerces-c_3 xerces-c_3_1 xerces-c-3.1 xerces-c_3_2 xerces-c-3.2 ${_XercesC_PATHS} PATH_SUFFIXES lib)
-    find_library(XercesC_LIBRARY_DEBUG NAMES xerces-c xerces-c_3D xerces-c_3_1D xerces-c-3.1D xerces-c_3_2D xerces-c-3.2D ${_XercesC_PATHS} PATH_SUFFIXES lib)
+    ## The NAMES_PER_DIR option will make sure that the PATHS are the "outer for loop" when searching for the libraries.
+	## We want that because we put the contrib as the first search path usually.
+    find_library(XercesC_LIBRARY_RELEASE NAMES xerces-c xerces-c_3 xerces-c_3_1 xerces-c-3.1 xerces-c_3_2 xerces-c-3.2 NAMES_PER_DIR ${_XercesC_PATHS} PATH_SUFFIXES lib)
```

Comment:
still look for the old versions?

Reply:
Yes otherwise we exclude most system contribs right?